core/frontend/src/tile/ThreeDTileFormatInterpreter.ts

```diff
@@ -0,0 +1,161 @@
+        Logger.logTrace(loggerCategory, "RealityData NOT Geolocated", () => ({ ...location }));
+      } else {
+        let ecefLocation: EcefLocation;
+        const locationOrientation = YawPitchRollAngles.tryFromTransform(worldToEcefTransform);
+        // Fix Bug 445630: [RDV][Regression] Orientation of georeferenced Reality Mesh is wrong.
+        // Use json.root.transform only if defined and not identity -> otherwise will use a transform computed from cartographic center.
+        if (!worldToEcefTransform.matrix.isIdentity && locationOrientation !== undefined && locationOrientation.angles !== undefined)
+          ecefLocation = new EcefLocation({ origin: locationOrientation.origin, orientation: locationOrientation.angles.toJSON() });
+        else
+          ecefLocation = EcefLocation.createFromCartographicOrigin(cartoCenter!);
+        location = ecefLocation;
+        Logger.logTrace(loggerCategory, "RealityData is worldToEcefTransform.matrix.isIdentity", () => ({ isIdentity: worldToEcefTransform!.matrix.isIdentity }));
+        // iModelDb.setEcefLocation(ecefLocation);
+        const ecefToWorld = ecefLocation.getTransform().inverse()!;
+        worldRange.extendRange(Range3d.fromJSON(ecefToWorld.multiplyRange(ecefRange)));
+        Logger.logTrace(loggerCategory, "RealityData ecefToWorld", () => ({ ...ecefToWorld }));
+      }
+    }
+    const spatialLocation: SpatialLocationAndExtents = { location, worldRange, isGeolocated };
+    return spatialLocation;
+  }
+  /** Gets information to identify the product and engine that create this reality data
+   * Will return undefined if cannot be resolved
+   * @param rootDocjson root document file in json format
+   * @returns information to identify the product and engine that create this reality data
+   * @alpha
+   */
+  public static getPublisherProductInfo(rootDocjson: any): PublisherProductInfo {
+    const info: PublisherProductInfo = {product: "", engine: "", version: ""};
+    if (rootDocjson && rootDocjson.root) {
+      if (rootDocjson.root.SMPublisherInfo) {
+        info.product = rootDocjson.root.SMPublisherInfo.Product ? rootDocjson.root.SMPublisherInfo.Product : "";
+        info.engine =  rootDocjson.root.SMPublisherInfo.Publisher ? rootDocjson.root.SMPublisherInfo.Publisher : "";
+        info.version = rootDocjson.root.SMPublisherInfo["Publisher Version"] ? rootDocjson.root.SMPublisherInfo["Publisher Version"] : "" ;
+      }
+    }
+    return info;
+  }
+  /** Gets information about 3dTile file for this reality data
+   * Will return undefined if cannot be resolved
+   * @param rootDocjson root document file in json format
+   * @returns information about 3dTile file for this reality data
+   * @internal
+   */
+  public static getFileInfo(rootDocjson: any): ThreeDTileFileInfo {
+    const info: ThreeDTileFileInfo = { rootChildren: 0};
+    if (rootDocjson && rootDocjson.root) {
+      info.rootChildren = parseInt(rootDocjson.root.children.length, 10);
```

Comment:
@MarcBedard8 Is `rootDocjson.root.children` an array? If so, `parseInt` is going to produce a run-time exception, because it expects a string.

Reply:
You're still not checking if `children` is `undefined`.
Change it to: 
```
const info: ThreeDTileFileInfo = {
  rootChildren: rootDocjson?.root?.children?.length ?? 0,
};
```

Reply:
Done